async-nats/src/options.rs

```diff
@@ -819,11 +819,20 @@ impl ConnectOptions {
         self
     }
 
+    /// By default, client construction will fail if
+    /// the connection to the server cannot be established.
+    ///
+    /// Setting `retry_on_initial_connect` causes client construction
```

Comment:
It's not exactly how it works. In this case, the handle to the `Client` is returned immediately and conection is handled asynchronously in the background.  Otherwise users would not know the reason for initial connection failure.

Reply:
I'm not sure I understand - if you don't mind, have a look again.